esp-wifi/src/wifi/mod.rs

```diff
@@ -340,6 +351,191 @@ impl Default for ClientConfiguration {
     }
 }
 
+#[cfg(csi_enable)]
+unsafe extern "C" fn csi_rx_cb(
+    ctx: *mut crate::wifi::c_types::c_void,
+    data: *mut crate::binary::include::wifi_csi_info_t,
+) {
+    let ctx = ctx as *mut alloc::boxed::Box<dyn FnMut(crate::wifi::wifi_csi_info_t)>;
```

Comment:
Your original type was `*mut dyn FnMut`, I'm not sure you should cast that to `*mut Box` 🤔

Reply:
I think this is correct, the type passed in to `set_receive_cb` is `Box<Box<dyn FnMut>>`. `set_receive_cb` calls `into_raw` to get a `*mut Box<dyn FnMut>`, which is what we're casting to here.

Reply:
Yeah apparently I missed one of the boxes. Alright